File: 2780. Minimum Index of a Valid Split/Solution.py
class Solution:
    def minimumIndex(self, nums: List[int]) -> int:
        d, x, high, n = dict(), 0, 0, len(nums)
        for num in nums:
            d[num] = 1 if num not in d else d[num] + 1
            if d[num] > high:
                high, x = d[num], num
        left, right = 0, high
        for i in range(len(nums) - 1):
            if nums[i] == x:
                left += 1
                right -= 1
            if left > (i + 1) // 2 and right > (n - i - 1) // 2: return i
        return -1

File: 2780. Minimum Index of a Valid Split/Solution.java
class Solution {
    public int minimumIndex(List<Integer> nums) {
        Map<Integer, Integer> d = new HashMap<>();
        int x = 0, high = 0, n = nums.size();
        for (int num : nums) {
            d.put(num, d.getOrDefault(num, 0) + 1);
            if (d.get(num) > high) {
                high = d.get(num);
                x = num;
            }
        }
        int left = 0, right = high;
        for (int i = 0; i < n - 1; i++) {
            if (nums.get(i) == x) {
                left++;
                right--;
            }
            if (left > (i + 1) / 2 && right > (n - i - 1) / 2) return i;
        }
        return -1;        
    }
}
